### PR TITLE
openai[patch]: add usage metadata details

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 1]
```diff
@@ -2160,3 +2154,34 @@ class OpenAIRefusalError(Exception):
...
+        ),
+        "reasoning": oai_token_usage.get("completion_tokens_details", {}).get(
+            "reasoning_tokens"
+        ),
+    }
+    return UsageMetadata(
+        input_tokens=input_tokens,
+        output_tokens=output_tokens,
+        total_tokens=total_tokens,
+        input_token_details=InputTokenDetails(
```
Comment (eyurtsev) [inline libs/partners/openai/langchain_openai/chat_models/base.py:2181]: Do you need InputTokenDetails here? Does it do a 2nd instantiation?

File: libs/partners/openai/tests/integration_tests/chat_models/test_base_standard.py [hunk 2]
```diff
@@ -20,3 +24,48 @@ def chat_model_params(self) -> dict:
...
+            "audio_output",
+            "reasoning_output",
+            "cache_read_input",
+            "cache_creation_input",
+        ]
+    ]:
+        return ["reasoning_output", "cache_read_input"]
+
+    def invoke_with_cache_read_input(self, *, stream: bool = False) -> AIMessage:
+        with open(REPO_ROOT_DIR / "README.md", "r") as f:
```
Comment (eyurtsev) [inline libs/partners/openai/tests/integration_tests/chat_models/test_base_standard.py:43]: Could we avoid loading a file here if ti's not necessary?

**Review by efriis — APPROVED** [review]

File: libs/partners/openai/tests/integration_tests/chat_models/test_base_standard.py (hunk 2 above)
Comment (baskaryan) [reply libs/partners/openai/tests/integration_tests/chat_models/test_base_standard.py:43]: having a >2k token prompt seems not great?

File: libs/partners/openai/langchain_openai/chat_models/base.py (hunk 1 above)
Comment (baskaryan) [reply libs/partners/openai/langchain_openai/chat_models/base.py:2181]: for lint, could type ignore

**Pull request merged**